### ISSUE TITLE
Feature Request

### ISSUE THREAD

**Issue by 9kartik** [rough-stuff/rough#202]
Cross hatch style sketchy linear gradient.

![image](https://user-images.githubusercontent.com/2690844/160063391-4e7210c3-543c-433b-90ce-02a454a3e94c.png)

This is the sketch style shading, currently we can look at linear gradients only which will just utilise hatching at different angles to mimic shading. 

One other variation would be stippling.

![image](https://user-images.githubusercontent.com/2690844/160063870-08e9a0ea-54a4-4395-9817-f06ac235588c.png)

Let me know if this would be a good feature to have, I'll be happy to contribute.

**Comment by pshihn** [comment]
Nice idea for sure. especially if it's done gradually rather than in blocks.
I am a bit hesitant to add another shader right now because I am envisioning that next major release will make shaders pluggable. So instead of having lots of shaders built in, people can use a specific shader as a separate module that plugs in.

**Comment by 9kartik** [comment]
Yeah, of course, the final implementation wouldn't be in blocks but based on length and angle user has selected.

Just want to know if the pluggable shaders are they going to be external libraries?

**Comment by perseveringman** [comment]
Is there any progress on support for gradient fills?
